geniusyield-market-maker/src/GeniusYield/MarketMaker/MakerBot.hs

```diff
@@ -115,19 +130,117 @@ buildAndSubmitActions user@User {uColl, uStakeCred} providers netId ua di = flip
     gyAwaitTxConfirmed providers awaitTxParams tid
     logInfo $ printf "Tx successfully seen on chain with %d confirmation(s)" numConfirms
 
-executeStrategy
-  :: Strategy
-  -> MakerBot
-  -> GYNetworkId
-  -> GYProviders
-  -> PricesProviders
-  -> DEXInfo
-  -> IO ()
-executeStrategy runStrategy MakerBot {mbUser, mbDelay, mbToken} netId providers pp di =
+evolveStrategy
+  ∷ Strategy
+  → MakerBot
+  → GYNetworkId
+  → GYProviders
+  → PricesProviders
+  → DEXInfo
+  → StateT MBFret IO ()
+evolveStrategy runStrategy mb@MakerBot {mbUser, mbDelay, mbToken} netId providers pp di = do
+  let cfg          = ppaCommonCfg . pricesAggregatorPP $ pp
+      respiteDelay = ceiling $ pccRespiteDelayFactor cfg * fromIntegral mbDelay
+  
   forever $ do
-    newActions ← runStrategy pp mbUser mbToken
+    mbFret ← get
+
+    case mbFret of
+      MBReady → do
+        (newActions, controller) ← lift $ runStrategy pp mbUser mbToken
+        case controller of
+          UACNormal     → do
+            lift $ buildAndSubmitActions mbUser providers netId newActions di
+
+            lift $ gyLogInfo providers logNS "---------- Done for the block! -----------"
+            lift $ threadDelay mbDelay
+
+          UACSpooked1   → do
+            put MBSpooked1 { mbs1Relax = 0, mbs1Worse = 0 }
+            lift $ gyLogInfo providers logNS "Closed all orders due to price mismatch among Prices Providers"
```

Comment:
Closing* as orders aren't yet closed.
Put a log again after `cancelAllOrders'` successfully returns.

Reply:
Done:  changed "Closed" to "Closing".  Function `cancelAllOrders'` already logs successful return with "No more orders to cancel!"